src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +346,55 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
                     }
                 }
 
+                string authenticatemethod = "maskinporten";
+
+                if (!string.IsNullOrEmpty(Request.Headers["X-Altinn-EnterpriseUser-Authentication"]))
+                {
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    byte[] decodedCredentials = Convert.FromBase64String(enterpriseUserHeader);
+                    string decodedString = Encoding.UTF8.GetString(decodedCredentials);
+                    string[] decodedStringArray = decodedString.Split(":");
+                    string usernameFromRequest = decodedStringArray[0];
+                    string password = decodedStringArray[1];
+
+                    string bridgeApiEndpoint = _generalSettings.BridgeAuthnApiEndpoint + "enterpriseuser";
+                    EnterpriseUserAuthenticationService enterpriseUserAuthenticationService = new EnterpriseUserAuthenticationService();
+
+                    EnterpriseUserCredentials credentials = new EnterpriseUserCredentials { UserName = usernameFromRequest, Password = password, OrganizationNumber = orgNumber };
+                    ConfiguredTaskAwaitable<HttpResponseMessage> response = enterpriseUserAuthenticationService.GetResponseMessage(credentials, bridgeApiEndpoint);
+
+                    var responseInfo = response.GetAwaiter().GetResult();
+                    string content = await response.GetAwaiter().GetResult().Content.ReadAsStringAsync();
+
+                    if (responseInfo.StatusCode.ToString() == "OK")
```

Comment:
Use the enums, eg `responseInfo.StatusCode == HttpStatusCode.OK`. Might wanna place the error handling on top, since they return immediately, and/or use a switch/case. This also needs handling for 404 from the bridge API, indicating that the user does not exist (or the password did not match). And it needs to deal with any other statuscode from bridge, eg. 500, which should result in a 502 Bad Gateway being returned to the client.

Reply:
Switched to switch-case format and started using the enums for HttpStatusCode.